### ISSUE TITLE
Unassigned descriptions should be approvable

### ISSUE THREAD

**Issue by sinabahram** [coyote-team/coyote#459]
I'm not suggesting a workflow difference e.g. resources still must go from assignment to approval, only that there is a shortcut option on unassigned descriptions that says approve. The software should then first assign the issue to whomever is hitting that option. This should only exist for admins or owner roles.

**Comment by flipsasser** [comment]
I'm a little concerned about the data model here - the only relationship a representation has to a user is the "author" column. Is the use case here is something like the following?

- Flip writes a description but it is not approved
- Sina comes along and approves the description

**Comment by sinabahram** [comment]
There are two issues at hand.

1, we could fix this by not allowing descriptions on an unassigned resource, full stop, but that's a bit harsh because it means for admins to just get some work done, they would constantly have to assign resources to themselves. That's "technically" right, but it's crappy UX.

2, So, if we can avoid doing 1, then what can we do? Well, my suggestion is that a user with sufficient permissions (either editor and higher or admin and higher) gets a describe button on unassigned rsources still, but others do not. So we partially do 1. However, then what do we do when that higher-privilege user hits describe? I think what should happen is when they enter a description and iht submit, at that moment (not when they hit the describe button, since they could close out), Coyote simply does the assignment for them e.g. here's the data flow.

user action:
Sina goes in as admin and so he gets a describe button on an unassigned resource. He hits it and then he puts in a description and hits submit.

data flow in system:
Sina assigns unassigned resource to Sina // this one is done by Coyote auto-magically
Sina describes now-assigned resource // that one happened for real

In short, this is simply a macro if you will, a shortcut, to make people's lives easier, but again, because of the breaking of the workflow, I don't even want to let editors do this because it's bad behavior. Admins, though, really need this when just setting up Coyote, testing things, playing with the API, etc. ... it's necessary.

Does that make sense, and can we make this happen?

**Comment by flipsasser** [comment]
Sure - can I ask, do you see an assignment as a prerequisite to a representation? The way I view it, an assignment is explicitly that: "Sina asked Flip to describe a resource." A representation, on the other hand, signifies something entirely different: "Flip described a resource."

If I'm understanding correctly, the use case you described would be, "Sina described a resource, which means Sina asked Sina to describe a resource in the act of describing it, but the moment that happened it was also completed."

If that's the case, I wonder if an assignment is a prerequisite to a representation, and if so why?

**Comment by flipsasser** [comment]
Yup, totally. I think that goes nicely with #485

**Comment by flipsasser** [comment]
That being said - your assertion that **only** admins should be able to describe resources without an assignment conflicts with #485, which is a little bit more flexible (editors, owners, OR admins can describe unassigned resources). Safe to assume "only admins" is a shorthand for those three roles, or should we need to revisit #485's requirements?

**Comment by sinabahram** [comment]
Sorry, nope, short-hand, and you’re absolutely correct to clarify that.

**Comment by flipsasser** [comment]
Fixed awhile back but never closed the issue
